omc3/utils/iotools.py

```diff
@@ -132,7 +175,26 @@ def remove_none_dict_entries(dict_: dict) -> dict:
     return {key: value for key, value in dict_.items() if value is not None}
 
 
-def save_config(output_dir: Path, opt: dict, script: str):
+def maybe_add_command(opt: dict, script: str) -> dict:
```

Comment:
Did not understand 😬

Reply:
Ah! This is actually a feature that came after a comment from you. And I am very proud of it.
This adds a new entry to the opt-dict called ";command" which contains the command used to run the current python script (from sys.argv). So this is similar to what was written in the `command.run` and you should be able to run this command and reproduce the current run/output. When the opt-dict is written out to an `.ini` the ";" at the beginning will cause this to be a comment in the ini, so you don't have to worry about issues when loading that file. But now you have the full run command in every saved ini file :)

Reply:
the "only if the executed file equals the given script." part means, that if you run your own script (`run_many_global_corrections.py`), which e.g. calls `global_correction` with different parameters, the command it not added in the case where `global_correction` calls this function, as the commandline parameters are probably completely different from the values written in the ini file.
So .. you can think of it a bit like `if __name__ == "__main__"` but a few levels deep.

Reply:
Very nice! I would suggest adding aaaaaaall of that text into the function docstring so it stays with the code ;)

Reply:
I KNEW you would say that xD

Reply:
You've already written it anyway, now it's mostly a copy paste :D